dissect/target/plugins/os/unix/log/journal.py

```diff
@@ -0,0 +1,457 @@
+    sd_id128_t  boot_id;
+    le64_t      xor_hash;
+    EntryItem_Compact   items[size - 64 / 4];
+};
+
+// The first four members are copied from from ObjectHeader, so that the size can be used as the length of entry_object_offsets
+struct EntryArrayObject {
+    ObjectType  type;
+    uint8_t     flags;
+    uint8_t     reserved[6];
+    le64_t      size;
+    le64_t      next_entry_array_offset;
+    le64_t      entry_object_offsets[size - 24 / 8];  // The size minus the previous members divided by the size of the offset
+};
+
+struct EntryArrayObject_Compact {
+    ObjectType  type;
+    uint8_t     flags;
+    uint8_t     reserved[6];
+    le64_t      size;
+    le64_t      next_entry_array_offset;
+    le32_t      entry_object_offsets[size - 24 / 4];
+};
+"""  # noqa: E501
+
+c_journal = cstruct()
+c_journal.load(journal_def)
+
+
+def get_optional(value: str, to_type: Callable):
+    """Return the value if True, otherwise return None."""
+    return to_type(value) if value else None
+
+
+class JournalFile:
+    """Parse Systemd Journal file format.
+
+    References:
+        - https://github.com/systemd/systemd/blob/206f0f397edf1144c63a158fb30f496c3e89f256/docs/JOURNAL_FILE_FORMAT.md
+        - https://github.com/libyal/dtformats/blob/c4fc2b8102702c64b58f145971821986bf74e6c0/documentation/Systemd%20journal%20file%20format.asciidoc
+    """  # noqa: E501
+
+    def __init__(self, fh: BinaryIO, target: Target):
+        self.fh = fh
+        self.target = target
+        self.header = c_journal.Header(self.fh)
+        self.signature = "".join(chr(c) for c in self.header.signature)
+        self.entry_array_offset = self.header.entry_array_offset
+
+    def entry_object_offsets(self) -> Iterator[int]:
+        """Read object entry arrays."""
+
+        offset = self.entry_array_offset
+
+        # Entry Array with next_entry_array_offset set to 0 is the last in the list
+        while offset != 0:
+            self.fh.seek(offset)
+
+            object = c_journal.ObjectHeader(self.fh)
+
+            if object.type == c_journal.ObjectType.OBJECT_ENTRY_ARRAY:
+                # After the object is checked, read again but with EntryArrayObject instead of ObjectHeader
+                self.fh.seek(offset)
+
+                if self.header.incompatible_flags & c_journal.IncompatibleFlag.HEADER_INCOMPATIBLE_COMPACT:
+                    entry_array_object = c_journal.EntryArrayObject_Compact(self.fh)
+                else:
+                    entry_array_object = c_journal.EntryArrayObject(self.fh)
+
+                for entry_object_offset in entry_array_object.entry_object_offsets:
+                    # Check if the offset is not zero and points to nothing
+                    if entry_object_offset:
+                        yield entry_object_offset
+
+                offset = entry_array_object.next_entry_array_offset
+
+    def decode_value(self, value: bytes) -> tuple[str, str]:
+        value = value.decode().strip()
```

Comment:
```suggestion
        value = value.decode(encoding="utf-8", errors="surrogateescape").strip()
```
If we do this change, certain errors can be prevented a we can parse more data:

![image](https://github.com/fox-it/dissect.target/assets/110607403/34291f09-b0c7-4394-bacf-f1f1b441e838)